mitmproxy/addonmanager.py

```diff
@@ -273,6 +274,38 @@ def invoke_addon(self, addon, event: hooks.Hook):
                         f"Addon handler {event.name} ({a}) not callable"
                     )
 
+    async def async_invoke_addon(self, addon, event: hooks.Hook):
+        """
+            Asynchronously invoke an event on an addon and all its children.
+        """
+        for addon, func in self._iter_hooks(addon, event):
+            res = func(*event.args())
+            # Support both async and sync hook functions
+            if isawaitable(res):
+                await res
+
+    def invoke_addon(self, addon, event: hooks.Hook):
+        """
+            Invoke an event on an addon and all its children.
+        """
+        for addon, func in self._iter_hooks(addon, event):
+            res = func(*event.args())
+            if isawaitable(res):
+                raise exceptions.AddonManagerError(
+                    f"Async handler {event.name} ({addon}) called from sync context"
+                )
+
+    async def async_trigger(self, event: hooks.Hook):
```

Comment:
In line with my previous comment, we eventually want everyone to call this method here instead of the sync `trigger()`.  This means we should give it a more permanent name that does not include async. How about we call this `trigger_event()`, then we can eventually add deprecation warnings for `trigger/invoke_addon` and tell people to migrate to this new (async) API?

Reply:
Do you have a suggested name for `invoke_addon`, then?

Reply:
`invoke_addon` is really only used by the script addon and the addon manager, happy to break API compatibility here. 🙊

Reply:
If you have nice other idea please feel free to go with it. I don't feel strongly about `invoke_addon` at all. :)